Tests/SwiftCliCoreTests/OffsetUtilityTest.swift

```diff
@@ -0,0 +1,169 @@
+        let expectedResult = 3
+        expect(result).to(equal(expectedResult))
+    }
+}
+class FindLeftDecTests: XCTestCase {
+
+    func testFindLeftDecForManyStitch() throws {
+        let result = OffsetUtility().findLeftChanges(stitchRow: ["k1", "k2tog", "k1", "p1", "k1", "p1", "k1"])
+        let expectedResult = -1
+        expect(result).to(equal(expectedResult))
+    }
+
+    func testFindLeftDecForOddRow() throws {
+        let result = OffsetUtility().findLeftChanges(stitchRow: ["k2tog", "k2tog", "k1", "p1", "k1", "p1", "k2tog"])
+        let expectedResult = -2
+        expect(result).to(equal(expectedResult))
+    }
+
+    func testFindLeftIncForOddRow() throws {
+        let result = OffsetUtility().findLeftChanges(stitchRow: ["k1", "yo", "k1", "p1", "k1", "p1", "k1"])
+        let expectedResult = 1
+        expect(result).to(equal(expectedResult))
+    }
+    func testFindLeftIncForEvenRow() throws {
+        let result = OffsetUtility().findLeftChanges(stitchRow: ["k1", "yo", "k1", "p1", "k1", "k1", "p1", "k1"])
+        let expectedResult = 1
+        expect(result).to(equal(expectedResult))
+    }
+
+}
+
+class FindRightDecTests: XCTestCase {
+
+    func testFindRightDecForManyStitch() throws {
+        let result = OffsetUtility().findRightChanges(stitchRow: ["k1", "k2tog", "k1", "p1", "k1", "p1", "k2tog"])
+        let expectedResult = -1
+        expect(result).to(equal(expectedResult))
+    }
+
+    func testFindRightDecForOddRow() throws {
+        let result = OffsetUtility().findRightChanges(stitchRow: ["k2tog", "k1", "k1", "k2tog", "k1", "p1", "k2tog"])
+        let expectedResult = -2
+        expect(result).to(equal(expectedResult))
+    }
+
+    func testFindRightIncForOddRow() throws {
+        let result = OffsetUtility().findRightChanges(stitchRow: ["k1", "yo", "k1", "p1", "k1", "p1", "yo", "k1"])
+        let expectedResult = 1
+        expect(result).to(equal(expectedResult))
+    }
+    func testFindRightIncForEvenRow() throws {
+        let result = OffsetUtility().findRightChanges(stitchRow: ["k1", "k1", "p1", "k1", "yo", "k1", "p1", "k1"])
+        let expectedResult = 1
+        expect(result).to(equal(expectedResult))
+    }
+
+}
+
+class MakeRowMetadataTests: XCTestCase {
+
+    func testFindRightDecForManyStith() throws {
+        let result = OffsetUtility().makeRowMetadata(stitchRow: ["k1", "yo", "k1", "p1", "k1", "p1", "yo", "k1"], rowNumber: 0)
+        let expectedResult = RowInfo(row: ["k1", "yo", "k1", "p1", "k1", "p1", "yo", "k1"],
+                                     rowNumber: 0,
+                                     bottomLine: "└─┴─┴─┴─┴─┴─┴─┴─┘",
+                                     stitchSymbols: "│ │o│ │-│ │-│o│ │\n",
+                                     width: 8,
+                                     leftIncDec: 1,
+                                     rightIncDec: 1,
+                                     leftOffset: 0)
+        expect(result).to(equal(expectedResult))
```

Comment:
A little surprised that you have only one test case for this, given that you have four `lazy var` logic to test, which I would separate out into individual cases.

Reply:
Adding new test suite for all lazy values